vernac/library.mli

```diff
@@ -41,13 +41,13 @@ type seg_proofs = Opaqueproof.opaque_proofterm array
     argument.
     [output_native_objects]: when producing vo objects, also compile the native-code version. *)
 
-type ('document,'counters) todo_proofs =
+type 'document todo_proofs =
  | ProofsTodoNone (* for .vo *)
  | ProofsTodoSomeEmpty of Future.UUIDSet.t (* for .vos *)
- | ProofsTodoSome of Future.UUIDSet.t * ((Future.UUID.t,'document) Stateid.request * bool) list * 'counters (* for .vio *)
+ | ProofsTodoSome of Future.UUIDSet.t * ((Future.UUID.t,'document) Stateid.request * bool) list (* for .vio *)
```

Comment:
This change looks fishy. I think you need to set some different "worker id" when you complete vio -> vo since if you restanrt from 0 and no worker id then you may collide with previously allocate univs (the dirpath is the same).

I'll sleep on it.

Reply:
The problem is real, see the test and the fix.